client/state/sites/plans/test/selectors.js

```diff
@@ -53,6 +55,74 @@ describe( 'selectors', () => {
 			expect( plans ).to.eql( plans1 );
 		} );
 	} );
+	describe( '#getCurrentPlan()', () => {
+		context( 'when no plan data is found for the given siteId', () => {
+			const siteId = 99999999;
+			const state = {
+				sites: {
+					plans: {
+						[ siteId ]: {}
+					}
+				}
+			};
+
+			it( 'returns null', () => {
+				const plan = getCurrentPlan( state, siteId );
+				expect( plan ).to.eql( null );
+			} );
+		} );
+
+		context( 'when plans are found for the given siteId', () => {
+			const siteId = 99999999;
+			const plan1 = { currentPlan: true };
+			const state = {
+				sites: {
+					plans: {
+						[ siteId ]: {
+							data: [ plan1 ]
+						}
+					},
+					items: {
+						[ siteId ]: {
+							URL: 'https://example.wordpress.com'
+						}
+					}
+				}
+			};
+
+			context( 'when those plans include a \'currentPlan\'', () => {
+				before( () => {
+					plan1.currentPlan = true;
+				} );
+
+				it( 'returns the currentPlan', () => {
+					const plan = getCurrentPlan( state, siteId );
+					expect( plan ).to.eql( plan1 );
+				} );
+			} );
+
+			context( 'when those plans do not include a \'currentPlan\'', () => {
+				let selectors, newSitePlanObject;
+
+				useMockery( ( mockery ) => {
+					mockery.registerMock( './assembler', {
+						createSitePlanObject: () => newSitePlanObject
```

Comment:
Is this mostly to keep the test data small here? In this case I think it'd be fine to also test the assembler. As any changes to it may cause this selector to work improperly.

For the most part in Calypso we use Mockery as a last resort when a library or subcomponent is problematic for unit testing. For example when the analytics library makes a call and assumes that window is always available. When attempting to test a component that uses that library we can then mock that to perform a noop, so we can write unit tests without having it error.

```javascript
mockery.registerMock( 'lib/analytics', { ga: { recordEvent: noop } } );
```